src/vm/fastserializer.cpp

```diff
@@ -11,7 +11,47 @@
 // As a result of work on V3 of Event Pipe (https://github.com/Microsoft/perfview/pull/532) it got removed
 // if you need it, please use git to restore it
 
-FastSerializer::FastSerializer(SString &outputFilePath)
+IpcStreamWriter::IpcStreamWriter(IpcStream *pStream) : _pStream(pStream)
+{
+    CONTRACTL
+    {
+        NOTHROW;
+        GC_TRIGGERS;
+        MODE_ANY;
+        PRECONDITION(_pStream != nullptr);
+    }
+    CONTRACTL_END;
+}
+
+IpcStreamWriter::~IpcStreamWriter()
+{
+    CONTRACTL
+    {
+        NOTHROW;
+        GC_TRIGGERS;
+        MODE_ANY;
+    }
+    CONTRACTL_END;
+
+    delete _pStream;
+}
+
+bool IpcStreamWriter::Write(const void *lpBuffer, const uint32_t nBytesToWrite, uint32_t &nBytesWritten) const
+{
+    CONTRACTL
+    {
+        NOTHROW;
+        GC_NOTRIGGER;
+        MODE_PREEMPTIVE;
+        PRECONDITION(lpBuffer != nullptr);
+        PRECONDITION(nBytesToWrite > 0);
+    }
+    CONTRACTL_END;
+
+    return _pStream->Write(lpBuffer, nBytesToWrite, nBytesWritten);
```

Comment:
This is the raw IPC stream right? Once the protocol is defined this thread is going to need cooperate with other threads that are also writing replies to other commands. This would include both locking synchronization and framing the data inside protocol packets. If that proves to be too onerous there is also Vance's proposal that the EventPipe data gets a dedicated channel, but that route would require some complexity in a different part of the code to set up the dedicated channel and associate it with the EventPipe session.

Reply:
Like above, I'm fine if you want to check this in as-is, then change it in the future once you want to pass a test that includes both events and command replies.

Reply:
Yes, that's the plan. Think of this as a file drop replacement or alternative :)

Reply:
we chatted about this offline, you were going to send an email with an updated plan for the protocol. IT renders my original comment moot.